sdk/communication/communication-jobrouter/review/communication-jobrouter.api.md

```diff
@@ -0,0 +1,807 @@
+export interface CompleteJobOptions extends JobRouterCompleteJobActionOptionalParams {
+    note?: string;
+}
+
+// @public
+export interface ConditionalQueueSelectorAttachment extends QueueSelectorAttachment {
+    condition: RouterRuleUnion;
+    kind: "conditional";
+    labelSelectors: QueueSelector[];
+}
+
+// @public
+export interface ConditionalWorkerSelectorAttachment extends WorkerSelectorAttachment {
+    condition: RouterRuleUnion;
+    kind: "conditional";
+    labelSelectors: WorkerSelector[];
+}
+
+// @public
+export interface CreateClassificationPolicyOptions extends JobRouterAdministrationUpsertClassificationPolicyOptionalParams {
+    fallbackQueueId?: string;
+    name?: string;
+    queueSelectors?: QueueSelectorAttachmentUnion[];
+}
+
+// @public
+export interface CreateDistributionPolicyOptions extends JobRouterAdministrationUpsertDistributionPolicyOptionalParams {
+    mode?: DistributionModeUnion;
+    name?: string;
+    offerTtlInSeconds?: number;
+}
+
+// @public
+export interface CreateExceptionPolicyOptions extends JobRouterAdministrationUpsertExceptionPolicyOptionalParams {
+    exceptionRules?: {
+        [propertyName: string]: ExceptionRule;
+    };
+    name?: string;
+}
+
+// @public
+export interface CreateJobOptions extends JobRouterUpsertJobOptionalParams {
+    channelId?: string;
+    channelReference?: string;
+    classificationPolicyId?: string;
+    dispositionCode?: string;
+    labels?: {
+        [propertyName: string]: any;
+    };
+    notes?: {
+        [propertyName: string]: string;
+    };
+    priority?: number;
+    queueId?: string;
+    requestedWorkerSelectors?: WorkerSelector[];
+    tags?: {
+        [propertyName: string]: any;
+    };
+}
+
+// @public
+export interface CreateQueueOptions extends JobRouterAdministrationUpsertQueueOptionalParams {
+    distributionPolicyId?: string;
+    exceptionPolicyId?: string;
+    labels?: {
+        [propertyName: string]: any;
+    };
+    name?: string;
+}
+
+// @public
+export interface CreateWorkerOptions extends JobRouterUpsertWorkerOptionalParams {
+    availableForOffers?: boolean;
+    channelConfigurations?: {
+        [propertyName: string]: ChannelConfiguration;
+    };
+    labels?: {
+        [propertyName: string]: any;
+    };
+    queueAssignments?: {
+        [propertyName: string]: Record<string, unknown>;
+    };
+    tags?: {
+        [propertyName: string]: any;
+    };
+    totalCapacity?: number;
+}
+
+// @public
+export interface DirectMapRule extends RouterRule {
+    kind: "direct-map-rule";
+}
+
+// @public
+export interface DistributionMode {
+    bypassSelectors?: boolean;
+    kind: "best-worker" | "longest-idle" | "round-robin";
+    maxConcurrentOffers: number;
+    minConcurrentOffers: number;
+}
+
+// @public (undocumented)
+export type DistributionModeUnion = DistributionMode | BestWorkerMode | LongestIdleMode | RoundRobinMode;
+
+// @public
+export interface DistributionPolicy {
+    readonly id?: string;
```

Comment:
It feels really weird that this id is optional, and leads customers to awkward situations where they need to eaither null check or use `!`. 

There is a good example of this in the samples
![image](https://user-images.githubusercontent.com/21109913/186249835-39a0840a-ec08-4c2c-b439-677a6dd9c731.png)

Is the service really not sending the id down in some scenarios? If this is ture there is not much that we can do, but if the service always sends an `id` we need to remove the `?` from it.

This applies to other interfaces where id is optional